### PR TITLE
Adding command BLMove

### PR DESCRIPTION
**Pull request by tjzhang-BQ**

### CODE REVIEW

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 1]
```diff
@@ -3832,6 +3833,44 @@ public T lmove(
         return getThis();
     }
 
+    /**
+     * Blocks the connection until it pops Atomically pops and removes the left/right-most element to
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:3837]:
```suggestion
     * Blocks the connection until it atomically pops and removes the left/right-most element to
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 2]
```diff
@@ -3832,6 +3833,44 @@ public T lmove(
...
+     * the list stored at <code>source</code> depending on <code>wherefrom</code>, and pushes the
+     * element at the first/last element of the list stored at <code>destination</code> depending on
+     * <code>wherefrom</code>. <code>BLMove</code> is the blocking variant of {@link #lmove(String,
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:3840]:
```suggestion
     * <code>wherefrom</code>. <br><code>BLMove</code> is the blocking variant of {@link #lmove(String,
```

**Review by acarbonetto — APPROVED** [review]
just a couple of minor documentation comments

**Comment by acarbonetto** [comment]
note: BLMOVE is already considered for timeout
https://github.com/aws/glide-for-redis/blob/main/glide-core/src/client/mod.rs#L196

**Comment by tjzhang-BQ** [comment]
> note: BLMOVE is already considered for timeout https://github.com/aws/glide-for-redis/blob/main/glide-core/src/client/mod.rs#L196

so if we encounter a command that is blocking and not yet included in `mod.rs` here we have to define the metrics for its timeout argument?  How is it so that commands that aren't done like this one already have its routing/timeout defined?

**Comment by Yury-Fridlyand** [comment]
All existing at the moment commands with timout added to `mod.rs` to handle them.

File: java/client/src/main/java/glide/api/commands/ListBaseCommands.java [hunk 3]
```diff
@@ -531,4 +531,49 @@ CompletableFuture<Map<String, String[]>> blmpop(
      */
     CompletableFuture<String> lmove(
             String source, String destination, ListDirection wherefrom, ListDirection whereto);
+
+    /**
+     * Blocks the connection until it pops atomically and removes the left/right-most element to the
+     * list stored at <code>source</code> depending on <code>wherefrom</code>, and pushes the element
+     * at the first/last element of the list stored at <code>destination</code> depending on <code>
+     * wherefrom</code>. <br>
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/ListBaseCommands.java:539]:
```suggestion
     * wherefrom</code>.<br>
```

File: java/client/src/main/java/glide/api/commands/ListBaseCommands.java [hunk 4]
```diff
@@ -531,4 +531,49 @@ CompletableFuture<Map<String, String[]>> blmpop(
...
+     * @param wherefrom The {@link ListDirection} the element should be removed from.
+     * @param whereto The {@link ListDirection} the element should be added to.
+     * @param timeout The number of seconds to wait for a blocking operation to complete. A value of
+     *     <code>0</code> will block indefinitely.
+     * @return The popped element or <code>null</code> if <code>source</code> does not exist.
+     * @example
+     *     <pre>{@code
+     * client.lpush("testKey1", new String[] {"two", "one"}).get();
+     * client.lpush("testKey2", new String[] {"four", "three"}).get();
+     * var result = client.blmove("testKey1", "testKey2", ListDirection.LEFT, ListDirection.LEFT).get();
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/ListBaseCommands.java:565]:
```suggestion
     * var result = client.blmove("testKey1", "testKey2", ListDirection.LEFT, ListDirection.LEFT, 0.1).get();
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 5]
```diff
@@ -3832,6 +3833,45 @@ public T lmove(
         return getThis();
     }
 
+    /**
+     * Blocks the connection until it atomically pops and removes the left/right-most element to the
+     * list stored at <code>source</code> depending on <code>wherefrom</code>, and pushes the element
+     * at the first/last element of the list stored at <code>destination</code> depending on <code>
+     * wherefrom</code>. <br>
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:3840]:
```suggestion
     * wherefrom</code>.<br>
```

File: java/integTest/src/test/java/glide/TransactionTestUtilities.java [hunk 6]
```diff
@@ -345,6 +349,10 @@ private static Object[] listCommands(BaseTransaction<?> transaction) {
                                 value1, // lmove(listKey6, listKey5, RIGHT, LEFT)
                                 new String[] {value2, value3}, // lrange(listKey6, 0, -1)
                                 new String[] {value3, value2, value1, value1}, // lrange(listKey7, 0, -1);
+                                value3, // .blmove(listKey7, listKey7, LEFT, LEFT, 0.1)
```
Comment (Yury-Fridlyand) [inline java/integTest/src/test/java/glide/TransactionTestUtilities.java:352]:
```suggestion
                                value3, // blmove(listKey7, listKey7, LEFT, LEFT, 0.1)
```

**Review by Yury-Fridlyand — APPROVED** [review]

**Pull request merged**